File: BE/src/main/java/com/example/model/service/RankingService.java
package com.example.model.service;

import com.example.model.dto.RankingDto;

import java.util.List;

public interface RankingService {

    // 상위 10명 랭킹 조회
    public List<RankingDto> getTopRankings();

    // 특정 유저의 랭킹 조회
    public RankingDto getUserRanking(int userId);

    RankingDto updateUserRanking(int userId, RankingDto rankingDto);
}

File: GameTset/app/src/main/java/com/example/gametset/room/data/remote/RankingService.kt
package com.example.gametset.ranking.data.remote

import com.example.gametset.ranking.data.model.dto.UserRankingDto
import retrofit2.http.GET
import retrofit2.http.Path

interface RankingService {
    @GET("ranking/{userId}")
    suspend fun getUserRanking(@Path("userId") userId: Int): UserRankingDto

    @GET("ranking/list")
    suspend fun getRankingList(): List<UserRankingDto>
}